dotnet/src/Extensions/PromptTemplates.Handlebars/Helpers/KernelHelpers/KernelSystemHelpers.cs

```diff
@@ -136,7 +136,7 @@ internal static class KernelSystemHelpers
                 return arg switch
                 {
                     bool booleanArg => booleanArg,
-                    _ => arg is null
+                    _ => arg is not null
```

Comment:
I think this change is correct but there is some issue in the tests for this helper that is causing two of the cases to fail. I am investigating those.

Reply:
Aligned offline - solution was a change to the `GetArgumentValue` method in `KernelHelperUtils.cs`.
 
Before, if the template referenced a variable that wasn't defined in KernelArguments, the method would just return the  key as a string. Now, it just returns null (as expected). 
 
I had it returning a string before because I thought Handlebars would default the argument to its string representation since it templates regular text, but I double checked [HB docs](https://handlebarsjs.com/guide/expressions.html#helpers), and generally, string literals are surrounded by quotes when the raw value is used as an argument

Reply:
Yes, I remember that I changed this code, because I was adding new tests to increase our code coverage to 80%, and there were no tests for `or` helper. When I added unit tests with expected outcome, it appeared that this helper was not working as expected, so I slightly changed it. 

The correct logic should be - if any of values in `or` operator is not null - then it should return true. So, this change is correct :)

Reply:
Awesome, thanks for the detail Dmytro!